test/js/api/domain/post/files-spec.js

```diff
@@ -0,0 +1,239 @@
+    };    
+            
+    it('should populate config for the default scenario', function() {
+        
+        var res = httpRes();
+        
+        var config = {
+            app: {data: defaultData},
+            api:{
+                defaultApi: {
+                    source: defaultSource
+                }
+            }
+        };
+                
+        respond(httpReq({api: defaultApi}), res, config);
+        assert.deepEqual(res._getData(), {  
+           scenarios:[  
+              'default'
+           ],
+           config:{  
+              api:'defaultApi',
+              name:'test_req',
+              file:'request_0'
+           },
+           default:[  
+              {  
+                 name:'a',
+                 file:'a'
+              },
+              {  
+                 name:'b',
+                 file:'b'
+              },
+              {  
+                 name:'c',
+                 file:'c'
+              }
+           ]
+        });        
+    });
+
+    it('should populate config for the provided scenario', function() {
+                
+        var res = httpRes();
+        
+        var config = {
+            app: {data: debugData},
+            api:{
+                debugApi: {
+                    source: debugSource,
+                    scenarios: {debugScenario: {source: debugSource}}
+                }
+            }
+        };
+                
+        respond(httpReq({api: debugApi}), res, config);        
+        assert.deepEqual(res._getData(), {  
+           scenarios:[  
+              'default',
+              'debugScenario'
+           ],
+           config:{  
+              api:'debugApi',
+              name:'test_req',
+              file:'request_0'
+           },
+           default:[  
+              {  
+                 name:'d',
+                 file:'d'
+              },
+              {  
+                 name:'e',
+                 file:'e'
+              },
+              {  
+                 name:'f',
+                 file:'f'
+              }
+           ],
+           debugScenario:[  
+              {  
+                 name:'d',
+                 file:'d'
+              },
+              {  
+                 name:'e',
+                 file:'e'
+              },
+              {  
+                 name:'f',
+                 file:'f'
+              }
+           ]
+        });           
+    });
+    
+    it('should skip an active scenario', function() {
+                
+        var res = httpRes();
+        
+        var config = {
+            app: {data: otherData},
+            api:{
+                otherApi: {
+                    source: otherSource,
+                    scenarios: {active: {source: otherSource}}
+                }
+            }
+        };
+                
+        respond(httpReq({api: otherApi}), res, config); 
+        assert.deepEqual(res._getData(), {  
+           scenarios:[  
```

Comment:
This holds the scenario's definitions. If it has a key called "active" that holds the key of the active scenario. There is no default scenario. If no scenario is available it just uses the global config. Each scenario is just an override for the global api config